insecure/net/network_egress_test.go

```diff
@@ -41,14 +41,14 @@ func TestHandleOutgoingEvent_AttackerObserve(t *testing.T) {
 		ccf)
 	require.NoError(t, err)
 
-	attacker := newMockAttackerObserverClient()
+	attacker := newMockAttacker()
 
 	attackerRegistered := sync.WaitGroup{}
 	attackerRegistered.Add(1)
 	go func() {
 		attackerRegistered.Done()
```

Comment:
I tried this but it doesn't work because `corruptNetwork.ConnectAttacker()` is a blocking call that waits for a shutdown signal, so the test times out if I use defer. The `attackerRegistered.Done()` has to be called before the blocking call.